src/textual/app.py

```diff
@@ -202,6 +209,105 @@ def exit(self, result: ReturnType | None = None) -> None:
         self._return_value = result
         self.close_messages_no_wait()
 
+    @property
+    def focus_chain(self) -> list[Widget]:
+        """Get widgets that may receive focus, in focus order."""
+        widgets: list[Widget] = []
+        add_widget = widgets.append
+        root = self.screen
+        stack: list[Iterator[Widget]] = [iter(root.displayed_children)]
+        pop = stack.pop
+        push = stack.append
+
+        while stack:
+            node = next(stack[-1], None)
+            if node is None:
+                pop()
+            else:
+                if node.is_container and node.can_focus:
+                    push(iter(node.displayed_children))
```

Comment:
Are we alright with invisible widgets getting focus? Can "focusable" be set dynamically on a Widget by users of it?

Reply:
It should probably filter `visible: hidden` out as well. Good point.

At the moment, its not designed to be dynamic. But maybe it should be, to skip over fields that have been disabled or something. I'll think about that for a future update.

Reply:
Added a `focusable_children` property.